src/utils/__tests__/internalLinks-test.js

```diff
@@ -0,0 +1,270 @@
+/* eslint-disable spellcheck/spell-checker */
+import {
+  isInternalLink,
+  isMessageLink,
+  getLinkType,
+  getNarrowFromLink,
+  getMessageIdFromLink,
+} from '../internalLinks';
+
+import { streamNarrow, topicNarrow, groupNarrow, specialNarrow } from '../narrow';
+
+describe('isInternalLink', () => {
+  test('when link is external, return false', () => {
+    expect(isInternalLink('https://example.com', 'https://another.com')).toBe(false);
+  });
+
+  test('when link is internal, but not in app, return false', () => {
+    expect(isInternalLink('https://example.com/user_uploads', 'https://example.com')).toBe(false);
+  });
+
+  test('when link is internal and in app, return true', () => {
+    expect(isInternalLink('https://example.com/#narrow/stream/jest', 'https://example.com')).toBe(
+      true,
+    );
+  });
+
+  test('when link is relative and in app, return true', () => {
+    expect(isInternalLink('#narrow/stream/jest/topic/topic1', 'https://example.com')).toBe(true);
+    expect(isInternalLink('/#narrow/stream/jest', 'https://example.com')).toBe(true);
+  });
+
+  test('links including IDs are also recognized', () => {
+    expect(isInternalLink('#narrow/stream/123-jest/topic/topic1', 'https://example.com')).toBe(
+      true,
+    );
+    expect(isInternalLink('/#narrow/stream/123-jest', 'https://example.com')).toBe(true);
+    expect(isInternalLink('/#narrow/pm-with/123-mark', 'https://example.com')).toBe(true);
+  });
+});
+
+describe('isMessageLink', () => {
+  test('only in-app link containing "near/<message-id>" is a message link', () => {
+    expect(isMessageLink('https://example.com/#narrow/stream/jest', 'https://example.com')).toBe(
+      false,
+    );
+    expect(isMessageLink('https://example.com/#narrow/#near/1', 'https://example.com')).toBe(true);
+  });
+});
+
+describe('getLinkType', () => {
+  test('links to a different domain are of "external" type', () => {
+    expect(getLinkType('https://google.com/some-path', 'https://example.com')).toBe('external');
+  });
+
+  test('only in-app link containing "stream" is a stream link', () => {
+    expect(
+      getLinkType('https://example.com/#narrow/pm-with/1,2-group', 'https://example.com'),
+    ).toBe('group');
+    expect(getLinkType('https://example.com/#narrow/stream/jest', 'https://example.com')).toBe(
+      'stream',
+    );
+    expect(getLinkType('https://example.com/#narrow/stream/stream/', 'https://example.com')).toBe(
+      'stream',
+    );
+  });
+
+  test('when a url is not a topic narrow return false', () => {
```

Comment:
This and some of the other test names should be updated in the commit that changes a lot of the test cases, "Start testing `getLinkType`" -- they no longer make sense as they are.